flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/utils/ParquetSchemaConverter.java

```diff
@@ -0,0 +1,359 @@
+			} else if (basicTypeInfo.equals(BasicTypeInfo.FLOAT_TYPE_INFO)) {
+				fieldType = Types.primitive(PrimitiveType.PrimitiveTypeName.FLOAT, repetition).named(fieldName);
+			} else if (basicTypeInfo.equals(BasicTypeInfo.LONG_TYPE_INFO)) {
+				fieldType = Types.primitive(PrimitiveType.PrimitiveTypeName.INT64, repetition).named(fieldName);
+			} else if (basicTypeInfo.equals(BasicTypeInfo.SHORT_TYPE_INFO)) {
+				fieldType = Types.primitive(PrimitiveType.PrimitiveTypeName.INT32, repetition).named(fieldName);
+			} else if (basicTypeInfo.equals(BasicTypeInfo.BYTE_TYPE_INFO)) {
+				fieldType = Types.primitive(PrimitiveType.PrimitiveTypeName.BINARY, repetition).named(fieldName);
+			} else if (basicTypeInfo.equals(BasicTypeInfo.CHAR_TYPE_INFO)) {
+				fieldType = Types.primitive(PrimitiveType.PrimitiveTypeName.BINARY, repetition).named(fieldName);
+			} else if (basicTypeInfo.equals(BasicTypeInfo.BOOLEAN_TYPE_INFO)) {
+				fieldType = Types.primitive(PrimitiveType.PrimitiveTypeName.BOOLEAN, repetition).named(fieldName);
+			} else if (basicTypeInfo.equals(BasicTypeInfo.DATE_TYPE_INFO)
+				|| basicTypeInfo.equals(BasicTypeInfo.STRING_TYPE_INFO)) {
+				fieldType = Types.primitive(PrimitiveType.PrimitiveTypeName.BINARY, repetition)
+					.as(OriginalType.UTF8)
+					.named(fieldName);
+			}
+		} else if (typeInfo instanceof MapTypeInfo) {
+			MapTypeInfo mapTypeInfo = (MapTypeInfo) typeInfo;
+			fieldType = Types.optionalMap()
+				.key(convertField(MAP_KEY, mapTypeInfo.getKeyTypeInfo(), Type.Repetition.REQUIRED, isStandard))
+				.value(convertField(MAP_VALUE, mapTypeInfo.getValueTypeInfo(), Type.Repetition.OPTIONAL, isStandard))
+				.named(fieldName);
+		} else if (typeInfo instanceof ObjectArrayTypeInfo) {
+			ObjectArrayTypeInfo objectArrayTypeInfo = (ObjectArrayTypeInfo) typeInfo;
+			fieldType = Types.optionalGroup()
+				.addField(convertField(LIST_ELEMENT, objectArrayTypeInfo.getComponentInfo(),
+					Type.Repetition.REPEATED, isStandard))
+				.as(OriginalType.LIST)
+				.named(fieldName);
+		} else if (typeInfo instanceof BasicArrayTypeInfo) {
+			BasicArrayTypeInfo basicArrayType = (BasicArrayTypeInfo) typeInfo;
+
+			if (isStandard) {
+
+				// Add extra layer of Group according to Parquet's standard
+				Type listGroup = Types.repeatedGroup().addField(
+					convertField(LIST_ELEMENT, basicArrayType.getComponentInfo(),
+						Type.Repetition.OPTIONAL, isStandard)).named(LIST_GROUP_NAME);
+
+				fieldType = Types.optionalGroup()
+					.addField(listGroup)
+					.as(OriginalType.LIST).named(fieldName);
+			} else {
+				PrimitiveType primitiveTyp =
+					convertField(fieldName, basicArrayType.getComponentInfo(),
+						Type.Repetition.OPTIONAL, isStandard).asPrimitiveType();
+				fieldType = Types.optionalGroup()
+					.repeated(primitiveTyp.getPrimitiveTypeName())
+					.as(primitiveTyp.getOriginalType())
+					.named(LIST_ARRAY_TYPE)
+					.as(OriginalType.LIST).named(fieldName);
+			}
+		} else {
```

Comment:
It seems that the time types are not handled. They do not belong to the basic types.